plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-tpm-eventlog-common.h"
+#include "fu-tpm-eventlog-device.h"
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_set_device_gtype (plugin, FU_TYPE_TPM_EVENTLOG_DEVICE);
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	const gchar *fn = "/sys/kernel/security/tpm0/binary_bios_measurements";
```

Comment:
Shouldn't this be udev based in case there are multiple tpms?    There should be uevents for subsystem `tpm` so you can at least match to correct character device.